internal/driver/responselistener.go

```diff
@@ -8,59 +8,17 @@ package driver
 
 import (
 	"encoding/json"
-	"fmt"
-	"net/url"
-	"strings"
-	"time"
 
 	"github.com/eclipse/paho.mqtt.golang"
 )
 
 func startCommandResponseListening() error {
-
-	var scheme = driver.serviceConfig.MQTTBrokerInfo.ResponseSchema
-	var brokerUrl = driver.serviceConfig.MQTTBrokerInfo.ResponseHost
-	var brokerPort = driver.serviceConfig.MQTTBrokerInfo.ResponsePort
-	var authMode = driver.serviceConfig.MQTTBrokerInfo.ResponseAuthMode
-	var secretPath = driver.serviceConfig.MQTTBrokerInfo.ResponseCredentialsPath
-	var mqttClientId = driver.serviceConfig.MQTTBrokerInfo.ResponseClientId
-	var qos = byte(driver.serviceConfig.MQTTBrokerInfo.ResponseQos)
-	var keepAlive = driver.serviceConfig.MQTTBrokerInfo.ResponseKeepAlive
+	var qos = byte(driver.serviceConfig.MQTTBrokerInfo.Qos)
 	var topic = driver.serviceConfig.MQTTBrokerInfo.ResponseTopic
 
-	uri := &url.URL{
-		Scheme: strings.ToLower(scheme),
-		Host:   fmt.Sprintf("%s:%d", brokerUrl, brokerPort),
-	}
-
-	err := SetCredentials(uri, "Response", authMode, secretPath)
-	if err != nil {
-		return err
-	}
-
-	var client mqtt.Client
-	for i := 1; i <= driver.serviceConfig.MQTTBrokerInfo.ConnEstablishingRetry; i++ {
-		client, err = createClient(mqttClientId, uri, keepAlive)
-		if err != nil && i == driver.serviceConfig.MQTTBrokerInfo.ConnEstablishingRetry {
-			return err
-		} else if err != nil {
-			driver.Logger.Error(fmt.Sprintf("Fail to initial conn for command response, %v ", err))
-			time.Sleep(time.Duration(driver.serviceConfig.MQTTBrokerInfo.ConnEstablishingRetry) * time.Second)
-			driver.Logger.Warn("Retry to initial conn for command response")
-			continue
-		}
-		break
-	}
-
-	defer func() {
-		if client.IsConnected() {
-			client.Disconnect(5000)
-		}
-	}()
-
-	token := client.Subscribe(topic, qos, onCommandResponseReceived)
+	token := driver.mqttClient.Subscribe(topic, qos, onCommandResponseReceived)
```

Comment:
Move subscribe to OnConnect handler so it re-subscribes on reconnects

Reply:
Modified.